packet_filter.py

```diff
@@ -64,15 +64,21 @@ def detect_tls_events(input_stream):
 
 
 def warn_about_ip_address(ip_address, bad_ips):
```

Comment:
It would be helpful to have object-oriented code. Ideally following Domain-Driven Design at some point.
I could imagine a `Rules` object with a `rules.match(ip)` method that returns the `Rule`s that it matches, which the client can then use to log or do whatever else.

The datasource could be ingested with a parse_rule that returns a `Rule`.
I can imagine code like `rules.add_rule(parse_rule(dataline))`

We should decouple the matching from the input format too.

Reply:
I shared this plan in a recent document sent to you, with RuleSet being the generalized representation for pattern matching against packets (and possibly streams in the future).  I thought it a little premature to do for the very simple matching we have with IPs but will definitely be a useful abstraction when reading from multiple IOC formats and overlapping rule components.

Reply:
I agree that abstractions are good but I also think not all abstractions are equally good, and hastily decided abstractions can affect velocity if they are at odds with later changes.  For example, your suggestion below to return a list of IOC feed names is at odds with the feeds that have a message defined for each rule, or for feeds that have more nuance than "these IPs are dangerous for any interaction".

I prefer to add abstractions when there are enough (or few but different enough) uses of the same code.  Otherwise, the abstraction tends to clutter the code and impede the reading of it.  Fortunately, I had been thinking about adding structure to this code for a while and was planning on adding it when adding the second IOC feed format we'll be reading from, and informed by a deeper understanding of .rules files and the kinds of evaluation/detection/reporting that we want, so rest assured the code organization you are asking for will be added here.